test/local_integration_test.py

```diff
@@ -0,0 +1,173 @@
+import boto3
+import logging
+from more_itertools import one
+import random
+import requests
+import time
+from typing import Set, Dict
+import unittest
+import urllib
+import uuid
+
+from azul import config
+from scripts.subscribe import subscribe
+from scripts.reindex import Reindexer
+
+
+logger = logging.getLogger(__name__)
+
+
+def setUpModule():
+    logging.basicConfig(level=logging.INFO)
+
+
+class IntegrationTest(unittest.TestCase):
+    bundle_uuid_prefix = None
+    prefix_length = 3
+
+    def setUp(self):
+        super().setUp()
+        self.set_lambda_test_mode(True)
+        self.bundle_uuid_prefix = ''.join([str(random.choice('abcdef0123456789')) for _ in range(self.prefix_length)])
+
+    def tearDown(self):
+        self.set_lambda_test_mode(False)
+        if config.subscribe_to_dss:
+            subscribe(config.dss_client(), subscribe=False)
```

Comment:
Wouldn't this permanently unsubscribe the environment the test is run in? I don't think that's desirable.

We want to test that subscribing and unsubscribing works but we don't want to change the subscription status of a deployment permanently.

If `config.subscribe_to_dss` is False you need to skip subscribing and unsubscribing. If it is True you want to do the equivalent of the code below as part of the test:

```
if config.subscribe_to_dss:
    unsubscribe()
    unsubscribe() # prove that is is idempotent
    subscribe()
```

and in tearDown():

```
    if config.subscribe_to_dss:
        subscribe() # prove that is is idempotent
```

This ensures that the deployment the test is run in remains subscribed to the DSS.